### PR TITLE
Refactor/110 http module loading process

### PR DESCRIPTION
**Pull request by sudiptob2**
Should we rename `VersionStrToSpecConfigMapping` ?

### CODE REVIEW

File: chk/modules/version/constants.py [hunk 1]
```diff
@@ -13,14 +12,3 @@ class VersionStrToSpecConfigMapping:
     data: Dict = {
         "default:http:0.7.2": "chk.modules.http.entities.HttpSpec_V072",
     }
```
Comment (0hsn) [inline chk/modules/version/constants.py:14]: this can be list, as we discussed. just to note.

**Review by 0hsn — COMMENTED** [review]
Looks awesome so far. let's continue this ...

File: chk/modules/version/constants.py (hunk 1 above)
Comment (sudiptob2) [reply chk/modules/version/constants.py:14]: yes I though about that, I has other dependencies, I will eventually refactor those

File: chk/modules/version/support.py [hunk 2]
```diff
@@ -50,6 +29,7 @@ def version_as_dict(self) -> dict[str, str]:
             raise SystemExit(err_message('fatal.V0005'))
 
         try:
-            return {key:self.document[key] for key in (VersionConfigNode.VERSION, ) if key in self.document}  # type: ignore
+            return {key: self.document[key] for key in (VersionConfigNode.VERSION,) if
+                    key in self.document}  # type: ignore
```
[the remark this answers is not in the document]
Comment (sudiptob2) [reply chk/modules/version/support.py:33]: uderstood

**Comment by sudiptob2** [comment]
@0hsn all tests are passing without any modification. Should we change the name of `VersionStrToSpecConfigMapping` It seems that this class is not providing the mapping to the execution unit anymore.

File: chk/modules/version/constants.py [hunk 3]
```diff
@@ -10,17 +9,4 @@ class VersionConfigNode:
...
-    def find_by_version(cls, key: str) -> str:
-        """find config by version str"""
-        if not isinstance(key, str): raise SystemExit(err_message('fatal.V0004'))
-
-        class_map = cls.data.get(key)
-        if not isinstance(class_map, str): raise SystemExit(err_message('fatal.V0004'))
-        elif len(class_map) == 0: raise SystemExit(err_message('fatal.V0004'))
-
-        return class_map
+    data: List = ['default:http:0.7.2']
```
Comment (0hsn) [inline chk/modules/version/constants.py:12]: rename from `data` to `request_versions`

File: chk/modules/version/constants.py [hunk 4]
```diff
@@ -10,17 +9,4 @@ class VersionConfigNode:
 class VersionStrToSpecConfigMapping:
```
Comment (0hsn) [inline chk/modules/version/constants.py:9]:
```suggestion
class VersionStore:
```

File: chk/modules/version/validation_rules.py [hunk 5]
```diff
@@ -8,6 +8,6 @@
         'required': True,
         'type': 'string',
         'empty': False,
-        'allowed': list(VersionStrToSpecConfigMapping.data.keys())
+        'allowed': VersionStrToSpecConfigMapping.data
```
Comment (0hsn) [inline chk/modules/version/validation_rules.py:11]: update name here `VersionStore`

Comment (0hsn) [inline chk/modules/version/validation_rules.py:11]:
```suggestion
        'allowed': VersionStore.data
```

**Review by 0hsn — CHANGES_REQUESTED** [review]
🙏

File: chk/modules/version/constants.py (hunk 4 above)
Comment (sudiptob2) [reply chk/modules/version/constants.py]: sure

**Review by 0hsn — APPROVED** [review]
Less code + static code path = good code

**Pull request merged**